python/HeaderService/hslib.py

```diff
@@ -343,11 +343,15 @@ def collect(self, keys):
     def collect_from_HeaderService(self):
         """Collect and update custom meta-data generated or transformed by the HeaderService"""
         self.DATE_OBS = hutils.get_date_utc(self.metadata['DATE-OBS'])
+        self.DATE_BEG = hutils.get_date_utc(self.metadata['DATE-BEG'])
         self.DATE_END = hutils.get_date_utc(self.metadata['DATE-END'])
         self.metadata['DATE-OBS'] = self.DATE_OBS.isot
+        self.metadata['DATE-BEG'] = self.DATE_BEG.isot
         self.metadata['DATE-END'] = self.DATE_END.isot
         self.metadata['DAYOBS'] = self.DATE_OBS.datetime.strftime('%Y%m%d')
```

Comment:
DAYOBS is not derived by the header service. It comes from the camera.  Maybe I'm misunderstanding this file.

Reply:
DAYOBS it is not coming from camera (at least at the moment) it is derived the startIntegration timeStamp that generates DATE-OBS.

Reply:
That can't be right. Camera has to publish all the components that form part of OBSID. Where do you get SEQNUM from? It's critical for the self-consistency of headers that all the components of OBSID and OBSID itself come from the same source of truth.  Note also that DAYOBS is not simply YYYMMDD from DATE-OBS because there is a UTC offset to account for the Chilean timezone.

Reply:
Camera is not publishing this yet, therefore the UGLY hack. This needs to be fix soon, once camera, which is the source of truth, publishes this too. See here for the ugly hack: https://github.com/lsst-dm/HeaderService/blob/tickets/DM-19258/python/HeaderService/hslib.py#L358

Reply:
Please fix DAYOBS in the same hacky way you fix SEQNUM.  You have no way of knowing the UTC offset that camera are using so if you are hacking OBSID up to get seqnum it's not a hardship to get the correct DAYOBS in the same way.

Reply:
OK -- done. Please remember that we need to fix this hack in the future.